dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -309,12 +305,14 @@ def login(new_sign_up_url = '')
     end
   end
 
-  private def register_new_user(user, new_sign_up_url = '')
+  private def register_new_user(user)
     PartialRegistration.persist_attributes(session, user)
```

Comment:
Just thought of something: in the new signup flow, don't we already know the user type by the time we get to the redirect? So why not just pull the user type in the controller instead of passing the URL (which only varies based on user type) via a cookie?

Reply:
We don't actually have user type on the back end- we only have it in session storage on the front end. I initially tried to grab @user.user_type but realized it isn't saved yet

Reply:
And we would still need another way to keep track of whether or not it's the new sign up flow, so would need to set some sort of cookie for that data

Reply:
But I think by the time the user even gets to the page where they see the oauth buttons, they've already selected a user type, right? Seems like it'd be simple enough to pass this information to the controller. For example, you could use a querystring param and set it in the Rails session, or use the PartialRegistration to cache it, then simply pull it when when you need to determine which page to render after the oauth redirect.

Reply:
For my own understanding and planning- would you consider this change in approach blocking?

Reply:
No I don't think the approach of cookie vs controller/cache is blocking; it's just my insight from the platform perspective and looking at how other similar logic is currently implemented. Regardless of which approach you decide to take, though, I think it'd be a significant improvement to store/pass `user_type` instead of the finish signup URL. Then the backend logic can handle switching on type and redirecting to the proper path, rather than redirecting to an arbitrary URL from the cookie. If we still need a separate indicator of new vs old flow that could be passed separately.

Reply:
@TurnerRiley can you foresee any challenges with passing the user_type param to the begin_sign_up registrations_controller post event? I'm working on it now but remembering that you mimicked the old action so as to alter those functions as little as possible

Reply:
It would take a bit of restructuring to make sure this only happens in the new flow but no big challenges come to mind! I'm also happy to pair on this since I imagine little signup things we don't immediately see now will pop up!

Reply:
I have a draft of it on this PR now- let's take a look at it together during today's pairing half hour!